src/vue/forms/UploadPreIssuanceForm.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+  <form
+    class="app-form issuance-form"
+    @submit.prevent="submit"
+  >
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <file-field
+          :label="'issuance.lbl-pre-issuance' | globalize"
+          :note="'issuance.lbl-file-type' | globalize"
+          accept=".iss"
+          v-model="documents.preIssuance"
+        />
+      </div>
+    </div>
+    <div
+      v-if="issuances.length"
+      class="pre-issuance-form__info"
+    >
+      <h3>{{ 'issuance.pre-issuance-info' | globalize }}</h3>
+      <table class="app__table">
+        <thead>
+          <th>{{ 'issuance.asset' | globalize }}</th>
+          <th>{{ 'issuance.amount' | globalize }}</th>
+        </thead>
+        <tbody>
+          <tr v-for="issuance in issuances" :key="issuance.asset">
+            <td>
+              {{ issuance.asset }}
+            </td>
+            <td>
+              {{
+                issuance.amount | formatMoney
+              }}
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+    <div class="app__form-actions">
+      <button
+        v-ripple
+        type="submit"
+        class="issuance-form__submit-btn"
+        :disabled="formMixin.isDisabled"
+      >
+        {{ 'issuance.upload' | globalize }}
+      </button>
+      <button
+        v-ripple
+        type="button"
+        class="issuance-form__cancel-btn"
+        @click.prevent="closeForm"
+      >
+        {{ 'issuance.cancel' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+import FileField from '@/vue/fields/FileField'
+
+import { Sdk } from '@/sdk'
+import { base } from '@tokend/js-sdk'
+import config from '@/config'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { FileHelper } from '@/js/helpers/file.helper'
+
+import { vuexTypes } from '@/vuex'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'create-issuance-form',
+  components: {
+    FileField
+  },
+  mixins: [FormMixin],
+  props: {
+    isShown: { type: Boolean, default: true }
+  },
+  data: _ => ({
+    documents: {
+      preIssuance: null
+    },
+    userOwnedTokens: null,
+    issuances: []
+  }),
+  computed: {
+    ...mapGetters([
+      vuexTypes.account
+    ])
+  },
+  watch: {
+    'documents.preIssuance': async function (value) {
+      if (value) {
+        const extracted = await FileHelper.readFileAsText(value.file)
+        try {
+          this.parsePreIssuances(JSON.parse(extracted).issuances)
```

Comment:
if we're uploading only first pre-issuance, why do we need to parse all of them?

Reply:
Resolved, fixed